.github/workflows/ci.ml

```diff
@@ -296,13 +301,18 @@ let main_build_job ~analyse_job ~cygwin_job ?section runner start_version ~oc ~w
   (* Intentionally fail fast, no need to run all build if there is a
    * problem in a given version; usually it is functions not defined in lower
    * versions of OCaml. *)
-  let (_fail_fast, matrix, _) = platform_ocaml_matrix ~fail_fast:true start_version in
   let (matrix, includes) =
     if platform = Windows then
-      (("host", ["x86_64-pc-cygwin"; "i686-w64-mingw32"; "x86_64-w64-mingw32"; "i686-pc-windows"; "x86_64-pc-windows"]) ::
-       ("build", ["x86_64-pc-cygwin"]) ::
-       matrix, [])
+      ([], [
+          [("host", "x86_64-pc-cygwin"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          [("host", "i686-w64-mingw32"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
```

Comment:
How about having the mingw-w64 i686 test OCaml 5.2.0 and the MSVC i686 test OCaml 4.14.2?

```suggestion
          [("host", "i686-w64-mingw32"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml5)];
```